CabalHelper/Compile.hs

```diff
@@ -413,6 +413,9 @@ errorInstallCabal cabalVer _distdir = panic $ printf "\
  where
    sver = showVersion cabalVer
 
+helperPrepared :: Version -> IO Bool
+helperPrepared compCabalVersion = isJust <$> cachedExe compCabalVersion
```

Comment:
I don't like that you're spreading the logic that's inside `compileHelper` around here.

In fact this introduces an inconsistency in the Cabal source tree corner case (`compileCabalSource`) since there isn't even a `cachedExe`, well there is but in a different place.